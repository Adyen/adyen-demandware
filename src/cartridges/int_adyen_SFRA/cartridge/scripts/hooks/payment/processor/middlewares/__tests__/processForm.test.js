/* eslint-disable global-require */
let processForm;
let req;
let paymentForm;
let viewFormData;

beforeEach(() => {
  processForm = require('../processForm');
  jest.clearAllMocks();
  req = {
    form: {
      adyenPaymentMethod: 'mockedPaymentMethod',
      brandCode: 'scheme',
    },
  };

  paymentForm = {
    paymentMethod: {
      htmlValue: 'mockedPaymentMethod',
    },
    creditCardFields: {
      cardType: {
        value: 'mockedCardType',
      },
      cardNumber: {
        value: 'mockedCardNumber',
      },
      saveCard: {
        checked: true,
      },
    },
    adyenPaymentFields: {
      adyenFingerprint: {
        value:
          'ryEGX8eZpJ0030000000000000KZbIQj6kzs0089146776cVB94iKzBGQGbvPiVrHq5S16Goh5Mk004ivbSuYdG0R00000YVxEr00000cru9sAxTR5wwQsLYbcA8:40',
      },
      adyenStateData: 'mockedStateData',
    },
  };

  viewFormData = {};
});

afterEach(() => {
  jest.resetModules();
});

describe('processForm', () => {
  it('should return error when credit card validation fails', () => {
    const processFormOutput = processForm(req, paymentForm, viewFormData);
    expect(processFormOutput).toMatchSnapshot();
  });

  it('should return viewData', () => {
    req.form.storedPaymentUUID = 'mockedUUID';
    const processFormOutput = processForm(req, paymentForm, viewFormData);
    expect(processFormOutput).toMatchSnapshot();
  });
});
