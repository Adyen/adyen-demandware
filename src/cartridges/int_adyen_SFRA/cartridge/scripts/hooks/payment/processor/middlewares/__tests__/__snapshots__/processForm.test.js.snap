// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processForm should return error when credit card validation fails 1`] = `
Object {
  "error": true,
  "fieldErrors": Object {
    "creditCardErrors": "mockedCreditCardErrors",
  },
}
`;

exports[`processForm should return viewData 1`] = `
Object {
  "error": false,
  "viewData": Object {
    "paymentInformation": Object {
      "adyenIssuerName": null,
      "adyenPaymentMethod": "mockedPaymentMethod",
      "cardNumber": "mockedCardNumber",
      "cardType": "mockedCardType",
      "isCreditCard": true,
      "stateData": undefined,
    },
    "paymentMethod": Object {
      "htmlName": undefined,
      "value": undefined,
    },
    "saveCard": true,
  },
}
`;
